lib/manageiq/cross_repo/runner.rb

```diff
@@ -5,20 +5,22 @@ module ManageIQ::CrossRepo
   class Runner
     attr_reader :test_repo, :core_repo, :gem_repos
 
-    def initialize(test_repo, core_repo, gem_repos)
+    def initialize(test_repo, repos)
       @test_repo = Repository.new(test_repo || "ManageIQ/manageiq@master")
+
+      core_repos, @gem_repos = Array(repos).collect { |repo| Repository.new(repo) }.partition(&:core?)
+      @core_repo = core_repos.first
+
       if @test_repo.core?
-        raise ArgumentError, "You cannot pass core repo when running a core test"           if core_repo.present? && core_repo != test_repo
-        raise ArgumentError, "You must pass at least one gem repo when running a core test" if gem_repos.blank?
+        raise ArgumentError, "You cannot pass core repo when running a core test"           if @core_repo.present? && @core_repo != @test_repo
+        raise ArgumentError, "You must pass at least one gem repo when running a core test" if @gem_repos.blank?
 
         @core_repo = @test_repo
       else
-        raise ArgumentError, "You must pass either a core repo or at least one gem repo when running a plugin test" if core_repo.blank? && gem_repos.blank?
+        raise ArgumentError, "You must pass either a core repo or at least one gem repo when running a plugin test" if repos.blank?
```

Comment:
I think this text should change to say "You must pass a least one repo when running a plugin test."

Reply:
:+1: done